### ISSUE TITLE
LogicException: A package can only be added to one repository

### ISSUE THREAD

**Issue by Soullivaneuh** [composer/composer#4643]
Since https://github.com/composer/composer/commit/6dbec8718d48bbf31b99ef91c7a01c349fc8f606, I got the following error on my [test file](https://github.com/Soullivaneuh/composer-versions-check/blob/aa873d4d05498512d40fc52548f3e8fa908ae6d7/tests/VersionsCheckTest.php#L120-L153):

```
LogicException: A package can only be added to one repository

/home/sullivan/projects/my/composer-versions-check/vendor/composer/composer/src/Composer/Package/BasePackage.php:130
/home/sullivan/projects/my/composer-versions-check/vendor/composer/composer/src/Composer/Repository/ArrayRepository.php:139
/home/sullivan/projects/my/composer-versions-check/vendor/composer/composer/src/Composer/Repository/ArrayRepository.php:35
/home/sullivan/projects/my/composer-versions-check/vendor/composer/composer/src/Composer/Command/DependsCommand.php:63
```

I'm completely stuck now. What should I change?

**Comment by Seldaek** [comment]
The commit you link is 4 years old.. so you're saying your plugin never worked? :p 

Anyway looking at your test case it seems you never add something twice to a repo so I don't get how this can fail sorry..

**Comment by Soullivaneuh** [comment]
Well, indeed, didn't notice that.

But I'm pretty sure to not have this issue before. Maybe a change but in another place.

Anyway, I found why: I use the depends command on a wrappe. This command rebuild packages on repository that was already done by my plugin. That's why I got this error.

I get rid of the error by not using the depends command and copy/paste the interessting code of this one.

Maybe this exception should not be thrown but a `Repository::addPackages` should ensure that the list is unique to make it more reusable?

Since issue is "fixed" for me, I let you appreciate if this issue should be closed or not. :+1: 

Thanks for your time.
